src/frontend/src/lib/components/tokens/TokenExchangeBalance.svelte

```diff
@@ -0,0 +1,16 @@
+<script lang="ts">
+	import { nonNullish } from '@dfinity/utils';
+	import type { TokenFinancialData } from '$lib/types/token';
+	import { formatUSD } from '$lib/utils/format.utils';
+
+	export let balance: TokenFinancialData['balance'];
+	export let usdBalance: TokenFinancialData['usdBalance'];
+</script>
+
+<output class="break-all">
+	{#if nonNullish(balance) && nonNullish(usdBalance)}
+		{formatUSD({ value: usdBalance })}
+	{:else}
+		{formatUSD({ value: 0, options: { minFraction: 0, maxFraction: 0 } }).replace('0', '-')}
```

Comment:
It's correct we want to display zero when not initialized, not a `&ZeroWidthSpace`? If correct, shouls it be animated `class:animate-pulse`?

Reply:
my understanding was that indeed: to display 0 not pulsating while the balance pulsates and loads. 

But you are right.... it doesn't make sense: I included pulsating, the same as balance.

It will pulsate ONLY when `balance` is nullish, as it was done before. That is because this component is used alsewhere and if `balance` is not nullish, the only case where `usdBalance` is nullish is when there are no exchange rates and in that case we already display `$-` in the rest of the code